doc/nrf/gs_modifying.rst

```diff
@@ -3,28 +3,51 @@
 Modifying a sample application
 ##############################
 
-After programming and testing a sample application, you probably want to make some modifications to the application, for example, change compilation options or add your own files with additional functionality.
+After programming and testing a sample application, you probably want to make some modifications to the application, for example, add your own files with additional functionality, change compilation options, or update the default configuration.
 
-Such changes must be included in the :file:`CMakeList.txt` file for the sample application.
-Instead of doing the changes manually in this file, you can use |SES| (SES) to  manage the project.
 
+Adding files and changing compiler settings
+*******************************************
 
-Adding files
-************
+All files that your application uses must be specified in the :file:`CMakeList.txt` file.
+By default, most samples include only the main application file :file:`src/main.c`.
+This means that you must add all other files that you are using.
 
-To add a file to a project, right-click **Project 'app/libapp.a'** in the Project Explorer.
-Select either **Add new file to CMakeLists.txt** to create a file and add it or **Add existing file to CMakeLists.txt** to add a file that already exists.
+You can also configure compiler options, application defines, or include directories in :file:`CMakeList.txt`.
+
+To update the :file:`CMakeList.txt` file, either edit it directly or use |SES| (SES) to maintain it.
+
+
+Editing :file:`CMakeList.txt` directly
+======================================
+
+Add all files that your application uses to the ``target_sources`` function in :file:`CMakeList.txt`.
+To include several files, it can be useful to specify them with a wildcard.
+For example, to include all ``.c`` files from the ``src`` folder, add the following lines to your :file:`CMakeList.txt`::
+
+   FILE(GLOB app_sources src/*.c)
+   target_sources(app PRIVATE ${app_sources})
+
+Instead of specifying each file (explicitly or with a wildcard), you can include all files from a folder by adding that folder as include folder::
+
+   zephyr_include_directories(src)
```

Comment:
This will add src to the global list of includes.

Often, the kernel and other subsystems don't include files in the app, so we will be increasing the scope of src more than necessary and unnecessarily increasing the risk of aliasing issues.

Instead we would usually want to do

target_include_directories(app PRIVATE src)